aequilibrae/project/network/gmns_builder.py

```diff
@@ -0,0 +1,523 @@
+import math
+import re
+import numpy as np
+import pandas as pd
+import string
+import shapely.wkb
+import shapely.wkt
+from shapely.geometry import LineString, Point
+from pyproj import Transformer
+from ...utils import WorkerThread
+
+from aequilibrae import logger
+from aequilibrae.parameters import Parameters
+
+
+class GMNSBuilder(WorkerThread):
+    def __init__(
+        self, net, link_path: str, node_path: str, uses_path: str = None, geom_path: str = None, srid: int = 4326
+    ) -> None:
+        WorkerThread.__init__(self, None)
+        self.p = Parameters()
+        self.links = net.links
+        self.nodes = net.nodes
+        self.link_types = net.link_types
+        self.modes = net.modes
+        self.conn = net.conn
+
+        self.link_df = pd.read_csv(link_path).fillna("")
+        self.node_df = pd.read_csv(node_path).fillna("")
+        self.uses_df = pd.read_csv(uses_path) if uses_path is not None else uses_path
+        self.geom_df = pd.read_csv(geom_path) if geom_path is not None else geom_path
+        self.srid = srid
+
+        self.l_equiv = self.p.parameters["network"]["gmns"]["link"]["equivalency"]
+        self.n_equiv = self.p.parameters["network"]["gmns"]["node"]["equivalency"]
+
+    def doWork(self):
+
+        p = self.p
+        gmns_n_fields = p.parameters["network"]["gmns"]["node"]["fields"]
+        gmns_l_fields = p.parameters["network"]["gmns"]["link"]["fields"]
+
+        # Checking if all required fields are in GMNS links and nodes files
+        for field in [x for x in gmns_n_fields if gmns_n_fields[x]["required"]]:
+            if field not in self.node_df.columns.to_list():
+                raise ValueError(f"In GMNS nodes file: field '{field}' required, but not found.")
+
+        for field in [x for x in gmns_l_fields if gmns_l_fields[x]["required"]]:
+            if field not in self.link_df.columns.to_list():
+                raise ValueError(f"In GMNS links file: field '{field}' required, but not found.")
+
+        gmns_geom = self.l_equiv["geometry"]
+        if gmns_geom not in self.link_df.columns.to_list():
+            if self.geom_df is None:
+                raise ValueError(
+                    "To create an aequilibrae links table, geometries information must be provided either in the GMNS link table or in a separate file ('geometry_path' attribute)."
+                )
+            else:
+                self.link_df = self.link_df.merge(self.geom_df, on="geometry_id", how="left")
+
+        # Checking if it is needed to change the spatial reference system.
```

Comment:
if it might be a no op maybe the name of the method could be `maybe_transform_srid` to be more explicit? that way it would be clear and the comment could be removed